astroimtools/cutout_tools.py

```diff
@@ -16,11 +16,270 @@
+
+    Examples
+    --------
+    If you have a list of hubble ultra deep field ra and dec coords,
+    you may use the tool as follows:
+        >>> from astropy.table import Table
+        >>> import astropy.units as u
+
+        >>> ra = [53.18782913, 53.14794797, 53.15059559] * u.deg
+        >>> dec = [-27.79405589, -27.77392421, -27.77158621] * u.deg
+        >>> ids = ["Galax_0", 123, 53.15059559*u.deg]
+        >>> cutout_width = cutout_height = [3.0, 4.0, 3.0] * u.arcsec
+
+        >>> catalog = Table(
+        ...     data=[ids, ra, dec, cutout_width, cutout_height],
+        ...     names=['id', 'ra', 'dec', 'cutout_width', 'cutout_height'])
+
+        # To get a list of PrimaryHDU objects:
+        >>> cutouts = cutout_tool('h_udf_wfc_b_drz_img.fits', catalog)
+        # To save to fits file, (returns a list of file names):
+        >>> cutouts = cutout_tool('h_udf_wfc_b_drz_img.fits', catalog, to_fits=True)
+
+        # If the above catalog table is saved in an ECSV file with the proper units information:
+        >>> cutouts = cutout_tool('h_udf_wfc_b_drz_img.fits', 'catalog.ecsv')
+
+    Parameters
+    ----------
+    image : str or `HDUList` or `PrimaryHDU` or `ImageHDU` or `CompImageHDU`
+        Image to cut from. If string is provided, it is assumed to be a
+        fits file path.
+    catalog : str or `astropy.table.table.Table`
+        Catalog table defining the sources to cut out. Must contain
+        unit information as the cutouttool does not assume default units.
+    image_ext : int, optional
+        Image extension to extract header and data. Default is 0.
+    origin : int
+        Whether pixel coordinates are 0 or 1-basedpixel coordinates.
+    to_fits : bool
+        Save cutouts to fits files. Each cutout will be saved to a separate file.
+    output_dir : str
+        Path to directory to save the cutouts in. Set to
+        current_working_dir/cutouts by default
+    overwrite: bool, optional
+        Overwrite existing files. Default is `False`.
+    catalog_format : str, BaseReader
+        Input catalog format if reading from a file. Default is `ascii.ecsv`.
+        Note: Catalog format should be capable of staring units as metadata.
+    delimiter: str
+        Input catalog column delimiter string if reading from a file.
+    suppress_rotation : bool
+        Suppress rotation even if ``'cutout_pa'`` is provided. Default is `False`.
+    verbose : bool, optional
+        Print extra info. Default is `True`.
+
+    Returns
+    -------
+    cutouts : list
+        A list of PrimaryHDU or file names (if saving to fits).
+        If cutout failed for a target, `None` will be added as a
+        place holder.
+    """
+    # Optional dependencies...
+    from reproject.interpolation.high_level import reproject_interp
+
+    # read in the catalog file:
+    if isinstance(catalog, str):
+        if delimiter is None:
+            catalog = QTable.read(catalog, format=catalog_format)
```

Comment:
As I discussed off-line with @robelgeda, I think it's better to *not* let the user specify the `format` - instead this should use the `Table.read`'s "guess" machinery, which is specifically designed to be flexible on what it accepts, but with a deterministic set of rules. We should instead tell the user that if they want control over how the `Table` is loaded, they have to use the `Table.read` options and pass in that table.  Otherwise we are duplicating code and giving two different ways to do the same thing.

One caveat here: it *is* reasonable for users to be able to expect it to work for either a fits table or an ascii table, though.  So it may be necessary to try both of those ways of doing it.  I think the `Table.read` machinery does that out-of-the-box, but should check just to make sure, and if not just work around it here by first trying a fits read and if it fails, falling back on `ascii`.